### PR TITLE
chore: add notebook tests

### PR DESCRIPTION
**Pull request by Hartorn**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

*Any change needs to be discussed before proceeding. Failure to do so may result in the rejection of the pull request.*

**What kind of change does this PR introduce?**
Add notebook tests

**Does this PR introduce a breaking change?**
no

**What needs to be documented once your changes are merged?**
nothing

**Closing issues**

Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,22 +1,176 @@
...
+                  POETRY_CACHE: /work/.cache/poetry
+                  PIP_CACHE_DIR: /work/.cache/pip
+                  JUPYTER_RUNTIME_DIR: /work/.cache/jupyter/runtime
+                  JUPYTER_CONFIG_DIR: /work/.cache/jupyter/config
+                  SHELL: bash -l
+        working_directory: /work
+        resource_class: large
+    docker-executor:
+        docker:
+            - image: dreamquark/docker:latest
```
Comment (Optimox) [inline .circleci/config.yml:22]: just to understand: is this where we name our executor or where we pull or image? What's specific with dreamquark/docker:latest image?

File: census_example.ipynb [hunk 2]
```diff
@@ -174,6 +174,15 @@
     "y_test = train[target].values[test_indices]"
    ]
   },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "max_epochs = 1000 if not os.getenv(\"CI\", False) else 20"
```
Comment (eduardocarvp) [inline census_example.ipynb:183]: Wouldn't 5 or 10 epochs be enough here? Just so the test can be faster :)

Comment (Optimox) [reply census_example.ipynb:183]: I think 5 or 10 would actually be enough for the tests, nothing specific happens between 5th and 20th epoch

File: .circleci/config.yml (hunk 1 above)
Comment (Hartorn) [reply .circleci/config.yml:22]: So we name the executor, with image name and so on, so we don't have to say each time which image use, where is the work dir and so on.

Dreamquark image is basically docker one, but installed with Make, and some other usefuls tools.
I can change this if you want in another PR, but then I need to install Make and so on. 
I think this is a bit faster, but I can change

**Pull request merged**